app/src/main/java/com/github/se/eventradar/ui/component/ReusableComponents.kt

```diff
@@ -376,6 +393,289 @@ fun CategoryDisplayColumn(
+    onValueChange: ((String) -> Unit),
+    errorState: Boolean
+) {
+  OutlinedTextField(
+      value = textFieldValue,
+      onValueChange = onValueChange,
+      label = { Text(textFieldLabel) },
+      modifier = modifier,
+      placeholder = { Text(placeHolderText) },
+      colors =
+          OutlinedTextFieldDefaults.colors()
+              .copy(
+                  focusedPrefixColor = MaterialTheme.colorScheme.primary,
+                  unfocusedPrefixColor = MaterialTheme.colorScheme.primary),
+      shape = RoundedCornerShape(12.dp),
+      isError = errorState,
+  )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DropdownInputTextField(
+    modifier: Modifier = Modifier,
+    textFieldLabel: String,
+    textFieldValue: String,
+    onValueChange: (String) -> Unit,
+    errorState: Boolean,
+    options: List<String>,
+    toggleIconTestTag: String,
+) {
+  var isExpanded by remember { mutableStateOf(false) }
+  var selectedOption by remember { mutableStateOf(textFieldValue) }
+
+  ExposedDropdownMenuBox(
+      expanded = isExpanded,
+      onExpandedChange = { isExpanded = !isExpanded },
+      modifier = Modifier.testTag("exposedDropDownMenuBox"),
+  ) {
+    OutlinedTextField(
+        value = selectedOption,
+        onValueChange = onValueChange,
+        label = { Text(textFieldLabel) },
+        modifier =
+            modifier.menuAnchor(
+                MenuAnchorType.PrimaryNotEditable), // Open dropdown on text field click
+        colors =
+            OutlinedTextFieldDefaults.colors()
+                .copy(
+                    focusedPrefixColor = MaterialTheme.colorScheme.primary,
+                    unfocusedPrefixColor = MaterialTheme.colorScheme.primary),
+        shape = RoundedCornerShape(12.dp),
+        isError = errorState,
+        trailingIcon = {
+          ExposedDropdownMenuDefaults.TrailingIcon(
+              expanded = isExpanded, modifier = Modifier.testTag(toggleIconTestTag))
+        },
+        readOnly = true, // Make the text field read-only to handle input via dropdown
+        singleLine = true)
+    ExposedDropdownMenu(
+        expanded = isExpanded,
+        onDismissRequest = { isExpanded = false },
+        modifier =
+            Modifier.exposedDropdownSize(true)
+                .testTag("exposedDropDownMenu")
+                .requiredSizeIn(maxHeight = 150.dp)) {
+          options.forEach { option ->
+            DropdownMenuItem(
+                text = { Text(text = option, fontSize = 16.sp) },
+                onClick = {
+                  selectedOption = option
+                  onValueChange(option)
+                  isExpanded = false
+                },
+                contentPadding = ExposedDropdownMenuDefaults.ItemContentPadding,
+            )
+          }
+        }
+  }
+}
+
+fun displayChosenOrganisers(organisers: List<String>): String {
+  var organisersString = ""
+  for (i in organisers.indices) {
+    if (i == 0) {
```

Comment:
more sophodtacted alternative: "organisersString += if (i == 0) {
          organisers[i]
      }"

Reply:
thank you for the suggestion, but i think that this style of coding is clearer & readable to me, which is why i kept it that way, let me know what you think